automated install/basic-install.sh

```diff
@@ -1399,6 +1399,9 @@ installConfigs() {
     install -T -m 0755 "${PI_HOLE_LOCAL_REPO}/advanced/Templates/pihole-FTL-poststop.sh" "${PI_HOLE_INSTALL_DIR}/pihole-FTL-poststop.sh"
 
     # If the user chose to install the dashboard,
+        # set permissions on /etc/lighttpd/lighttpd.conf so pihole user (other) can read the file
+        chmod o+x /etc/lighttpd
+        chmod o+r "${lighttpdConfig}"
     if [[ "${INSTALL_WEB_SERVER}" == true ]]; then
```

Comment:
Got swapped when resolving the conflict I guess.
```suggestion
    if [[ "${INSTALL_WEB_SERVER}" == true ]]; then
        # set permissions on /etc/lighttpd/lighttpd.conf so pihole user (other) can read the file
        chmod o+x /etc/lighttpd
        chmod o+r "${lighttpdConfig}"
```
Btw, why does `pihole` need to read the Lighttpd config file?

Reply:
If it needs to read it, we should consider adding `pihole` to the `lighttpd` group rather than adding access for `other` (which means a lot more than the user `pihole` alone)

Reply:
Back at the computer, I was able to find where reading is done: In the debugging script.

https://github.com/pi-hole/pi-hole/blob/21026d941428bfcb4f405479db470a90bae6aec1/advanced/Scripts/piholeDebug.sh#L68

https://github.com/pi-hole/pi-hole/blob/21026d941428bfcb4f405479db470a90bae6aec1/advanced/Scripts/piholeDebug.sh#L78

https://github.com/pi-hole/pi-hole/blob/21026d941428bfcb4f405479db470a90bae6aec1/advanced/Scripts/piholeDebug.sh#L140-L141

https://github.com/pi-hole/pi-hole/blob/21026d941428bfcb4f405479db470a90bae6aec1/advanced/Scripts/piholeDebug.sh#L1095-L1114

Reply:
Ah right, and that can be done from web interface as well.

The problem with using group permissions is that `/etc/lighttpd` and content is usually owned by `root`, not by the Lighttpd service user. So it would require `pihole` to be a member of the `root` group, which grants it much more permission than we want.

Since it is unlikely that Lighttpd itself runs as root, this PR obviously is to cover cases where the dir/content is not owned by `root` group, but then necessarily by the Lighttpd service user/group. Otherwise without `o=x/r` it wouldn't be able to start. So in theory, the `pihole` user could be added to the owner group of `/etc/lighttpd` and `/etc/lighttpd/lighttpd.conf` only if it is not root, otherwise we can rely on it being world-readable. However, this adds quite some complexity, including checking groups of both, the dir and the config file, and setting `g=x/r` to be failsafe, not sure whether this is worth it 🤔.

Reply:
This PR only sets the same permissions as we did before

https://github.com/pi-hole/pi-hole/blob/8d2e023ec0a2c52c309fd7971d76eeaef124ad65/automated%20install/basic-install.sh#L1391

https://github.com/pi-hole/pi-hole/blob/8d2e023ec0a2c52c309fd7971d76eeaef124ad65/automated%20install/basic-install.sh#L1398

or will do in the future

https://github.com/pi-hole/pi-hole/blob/955e36a95573260ddfabf7943e805c57450a5030/automated%20install/basic-install.sh#L1405
https://github.com/pi-hole/pi-hole/blob/955e36a95573260ddfabf7943e805c57450a5030/automated%20install/basic-install.sh#L1416